modules/core/shared/src/main/scala/gem/util/InstantMicros.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.util
+
+import cats._
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit.MICROS
+
+
+/** InstantMicros wraps a `java.util.Instant` that is truncated to microsecond
+  * resolution.  This allows InstantMicros to roundtrip to/from the database
+  * where timestamps support only microsecond resolution.
+  *
+  * @param toInstant
+  */
+final class InstantMicros private (val toInstant: Instant) extends AnyVal {
```

Comment:
The `private` just hides the constructor but the `val` makes `toInstant` public.  I've always thought that syntax to be a bit awkward.  Here I wanted to hide the constructor because not just any `Instant` can be used -- it has to be a truncated instant.